src/MongoDB.Driver.Core/MongoDB.Driver.Core.csproj

```diff
@@ -25,6 +25,18 @@
     <IncludeSymbols>true</IncludeSymbols>
   </PropertyGroup>
 
+  <PropertyGroup>
+    <IsWindows Condition="$([MSBuild]::IsOSPlatform('Windows'))">true</IsWindows>
```

Comment:
The main change here where we determine whether the current platform is windows or no